src/main/java/com/google/step/model/Deal.java

```diff
@@ -28,4 +28,29 @@ public Deal(
     this.posterId = posterId;
     this.restaurantId = restaurantId;
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null || !(obj instanceof Deal)) {
+      return false;
+    }
+    Deal other = (Deal) obj;
+    if (other == this) {
+      return true;
+    }
+
+    return (other.id == this.id)
+        && ((other.description == null && this.description == null)
+            || (this.description != null && this.description.equals(other.description)))
```

Comment:
Consider having a helper method in the model package to help with this nullable String comparison, I think you can replace a lot of duplicate code in the model classes this way.